# 模型训练和评估验证
* 数据来源：../datasets/data_feature_processed.csv
* 代码内容：训练xgboost模型，并进行线下评估

# 导入包

In [5]:
import xgboost as xgb
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error
import pandas as pd
import numpy as np
import time
import re
%matplotlib inline

# 导入特征工程后的数据

In [6]:
inputfile = '../datasets/data_feature_processed.csv'
modelfile = '../model/xgb.pkl'

In [7]:
data = pd.read_csv(inputfile)

# XGBoost模型训练

In [9]:
df = data.copy()
df.sort_values(by=['loc_id','time_stamp'],inplace=True)

## 1. 划分训练集和测试集

In [42]:
train = df[df.time_stamp<'2017-10-01']
test = df[df.time_stamp>='2017-10-01']
train_X = train.iloc[:,3:]
train_y = train['num_of_people']
test_X = test.iloc[:,3:]
test_y = test['num_of_people']
print(train_X.shape,test_X.shape)

(191664, 90) (18216, 90)


## 2. XGBoost训练

In [43]:
model_xgb = xgboost_model_train(train_X,train_y,test_X,test_y)

[0]	train-rmse:594.044	test-rmse:1038.57
Multiple eval metrics have been passed: 'test-rmse' will be used for early stopping.

Will train until test-rmse hasn't improved in 20 rounds.
[1]	train-rmse:577.615	test-rmse:1011.55
[2]	train-rmse:561.494	test-rmse:985.231
[3]	train-rmse:545.966	test-rmse:959.46
[4]	train-rmse:530.817	test-rmse:934.149
[5]	train-rmse:516.123	test-rmse:909.949
[6]	train-rmse:501.948	test-rmse:886.75
[7]	train-rmse:488.159	test-rmse:863.722
[8]	train-rmse:474.848	test-rmse:841.794
[9]	train-rmse:462.169	test-rmse:821.043
[10]	train-rmse:449.603	test-rmse:800.743
[11]	train-rmse:437.695	test-rmse:781.336
[12]	train-rmse:425.886	test-rmse:761.875
[13]	train-rmse:414.689	test-rmse:743.05
[14]	train-rmse:403.776	test-rmse:724.929
[15]	train-rmse:393.045	test-rmse:707.286
[16]	train-rmse:382.646	test-rmse:689.634
[17]	train-rmse:372.777	test-rmse:672.284
[18]	train-rmse:363.172	test-rmse:656.029
[19]	train-rmse:353.794	test-rmse:640.827
[20]	train-rmse:344.621	test-r

[191]	train-rmse:87.4698	test-rmse:203.643
[192]	train-rmse:87.3849	test-rmse:203.644
[193]	train-rmse:87.311	test-rmse:203.596
[194]	train-rmse:87.2734	test-rmse:203.544
[195]	train-rmse:87.1741	test-rmse:203.454
[196]	train-rmse:87.0995	test-rmse:203.43
[197]	train-rmse:87.002	test-rmse:203.315
[198]	train-rmse:86.9296	test-rmse:203.278
[199]	train-rmse:86.8816	test-rmse:203.252
[200]	train-rmse:86.8186	test-rmse:203.228
[201]	train-rmse:86.759	test-rmse:203.271
[202]	train-rmse:86.681	test-rmse:203.233
[203]	train-rmse:86.5949	test-rmse:203.123
[204]	train-rmse:86.5004	test-rmse:203.098
[205]	train-rmse:86.4144	test-rmse:203.084
[206]	train-rmse:86.3432	test-rmse:202.955
[207]	train-rmse:86.26	test-rmse:202.898
[208]	train-rmse:86.206	test-rmse:202.89
[209]	train-rmse:86.1771	test-rmse:202.924
[210]	train-rmse:86.1101	test-rmse:202.857
[211]	train-rmse:86.0479	test-rmse:202.846
[212]	train-rmse:85.9822	test-rmse:202.78
[213]	train-rmse:85.9427	test-rmse:202.791
[214]	train-rmse:85.8

[383]	train-rmse:78.2701	test-rmse:200.277
[384]	train-rmse:78.2332	test-rmse:200.278
[385]	train-rmse:78.2002	test-rmse:200.302
[386]	train-rmse:78.1361	test-rmse:200.288
[387]	train-rmse:78.0941	test-rmse:200.286
[388]	train-rmse:78.0873	test-rmse:200.284
[389]	train-rmse:78.0125	test-rmse:200.286
[390]	train-rmse:77.9744	test-rmse:200.258
[391]	train-rmse:77.9134	test-rmse:200.242
[392]	train-rmse:77.8919	test-rmse:200.246
[393]	train-rmse:77.8336	test-rmse:200.204
[394]	train-rmse:77.7555	test-rmse:200.114
[395]	train-rmse:77.7423	test-rmse:200.132
[396]	train-rmse:77.665	test-rmse:200.077
[397]	train-rmse:77.6374	test-rmse:200.066
[398]	train-rmse:77.6062	test-rmse:200.055
[399]	train-rmse:77.5793	test-rmse:200.058
[400]	train-rmse:77.5286	test-rmse:200.062
[401]	train-rmse:77.5062	test-rmse:200.084
[402]	train-rmse:77.4604	test-rmse:200.069
[403]	train-rmse:77.4283	test-rmse:200.073
[404]	train-rmse:77.4144	test-rmse:200.088
[405]	train-rmse:77.4088	test-rmse:200.109
[406]	train-

## 3. 模型评估
* 使用10月份数据进行实时预测

In [47]:
Oct_test_df = pd.read_csv('../datasets/Oct_test_data.csv',parse_dates=['time_stamp'])

In [48]:
%time pred_df = real_time_predict(model_xgb,Oct_test_df)

D:\Anaconda3\lib\site-packages\ipykernel_launcher.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


[180.]
第0行预测完成！
[129.]
第1行预测完成！
[110.]
第2行预测完成！
[109.]
第3行预测完成！
[116.]
第4行预测完成！
[116.]
第5行预测完成！
[167.]
第6行预测完成！
[401.]
第7行预测完成！
[617.]
第8行预测完成！
[662.]
第9行预测完成！
[511.]
第10行预测完成！
[775.]
第11行预测完成！
[1025.]
第12行预测完成！
[807.]
第13行预测完成！
[572.]
第14行预测完成！
[510.]
第15行预测完成！
[511.]
第16行预测完成！
[805.]
第17行预测完成！
[814.]
第18行预测完成！
[562.]
第19行预测完成！
[526.]
第20行预测完成！
[535.]
第21行预测完成！
[562.]
第22行预测完成！
[351.]
第23行预测完成！
[188.]
第24行预测完成！
[143.]
第25行预测完成！
[129.]
第26行预测完成！
[139.]
第27行预测完成！
[146.]
第28行预测完成！
[141.]
第29行预测完成！
[180.]
第30行预测完成！
[498.]
第31行预测完成！
[737.]
第32行预测完成！
[683.]
第33行预测完成！
[527.]
第34行预测完成！
[820.]
第35行预测完成！
[1020.]
第36行预测完成！
[775.]
第37行预测完成！
[575.]
第38行预测完成！
[525.]
第39行预测完成！
[503.]
第40行预测完成！
[841.]
第41行预测完成！
[806.]
第42行预测完成！
[567.]
第43行预测完成！
[526.]
第44行预测完成！
[539.]
第45行预测完成！
[541.]
第46行预测完成！
[343.]
第47行预测完成！
[182.]
第48行预测完成！
[139.]
第49行预测完成！
[119.]
第50行预测完成！
[119.]
第51行预测完成！
[121.]
第52行预测完成！
[118.]
第53行预测完成！
[191.]
第54行预测完成！
[546.]
第55行预测完成！
[748.]
第56行预测完成！
[656.]
第57行预测完成！
[503.]
第58行预测完成！
[823.

[825.]
第467行预测完成！
[935.]
第468行预测完成！
[687.]
第469行预测完成！
[565.]
第470行预测完成！
[529.]
第471行预测完成！
[514.]
第472行预测完成！
[780.]
第473行预测完成！
[784.]
第474行预测完成！
[538.]
第475行预测完成！
[476.]
第476行预测完成！
[449.]
第477行预测完成！
[441.]
第478行预测完成！
[257.]
第479行预测完成！
[157.]
第480行预测完成！
[120.]
第481行预测完成！
[108.]
第482行预测完成！
[99.]
第483行预测完成！
[97.]
第484行预测完成！
[104.]
第485行预测完成！
[176.]
第486行预测完成！
[319.]
第487行预测完成！
[470.]
第488行预测完成！
[570.]
第489行预测完成！
[512.]
第490行预测完成！
[779.]
第491行预测完成！
[867.]
第492行预测完成！
[625.]
第493行预测完成！
[531.]
第494行预测完成！
[507.]
第495行预测完成！
[518.]
第496行预测完成！
[738.]
第497行预测完成！
[762.]
第498行预测完成！
[525.]
第499行预测完成！
[472.]
第500行预测完成！
[460.]
第501行预测完成！
[454.]
第502行预测完成！
[268.]
第503行预测完成！
[171.]
第504行预测完成！
[141.]
第505行预测完成！
[127.]
第506行预测完成！
[122.]
第507行预测完成！
[116.]
第508行预测完成！
[121.]
第509行预测完成！
[196.]
第510行预测完成！
[528.]
第511行预测完成！
[784.]
第512行预测完成！
[707.]
第513行预测完成！
[558.]
第514行预测完成！
[945.]
第515行预测完成！
[1100.]
第516行预测完成！
[853.]
第517行预测完成！
[680.]
第518行预测完成！
[602.]
第519行预测完成！
[587.]
第520行预测完成！
[1035.]
第521行预测完成！
[974.]
第52

第380行预测完成！
[601.]
第381行预测完成！
[617.]
第382行预测完成！
[359.]
第383行预测完成！
[203.]
第384行预测完成！
[198.]
第385行预测完成！
[210.]
第386行预测完成！
[208.]
第387行预测完成！
[209.]
第388行预测完成！
[209.]
第389行预测完成！
[267.]
第390行预测完成！
[626.]
第391行预测完成！
[759.]
第392行预测完成！
[634.]
第393行预测完成！
[501.]
第394行预测完成！
[883.]
第395行预测完成！
[1158.]
第396行预测完成！
[840.]
第397行预测完成！
[650.]
第398行预测完成！
[589.]
第399行预测完成！
[546.]
第400行预测完成！
[1011.]
第401行预测完成！
[956.]
第402行预测完成！
[641.]
第403行预测完成！
[581.]
第404行预测完成！
[603.]
第405行预测完成！
[616.]
第406行预测完成！
[367.]
第407行预测完成！
[200.]
第408行预测完成！
[200.]
第409行预测完成！
[211.]
第410行预测完成！
[207.]
第411行预测完成！
[205.]
第412行预测完成！
[206.]
第413行预测完成！
[270.]
第414行预测完成！
[613.]
第415行预测完成！
[766.]
第416行预测完成！
[615.]
第417行预测完成！
[492.]
第418行预测完成！
[871.]
第419行预测完成！
[1186.]
第420行预测完成！
[846.]
第421行预测完成！
[649.]
第422行预测完成！
[592.]
第423行预测完成！
[565.]
第424行预测完成！
[998.]
第425行预测完成！
[965.]
第426行预测完成！
[643.]
第427行预测完成！
[571.]
第428行预测完成！
[600.]
第429行预测完成！
[608.]
第430行预测完成！
[360.]
第431行预测完成！
[191.]
第432行预测完成！
[186.]
第433行预测完成！
[182.]
第434行预测完成！
[183.]
第435行预测

第280行预测完成！
[1762.]
第281行预测完成！
[1830.]
第282行预测完成！
[1258.]
第283行预测完成！
[1043.]
第284行预测完成！
[1065.]
第285行预测完成！
[961.]
第286行预测完成！
[384.]
第287行预测完成！
[120.]
第288行预测完成！
[88.]
第289行预测完成！
[80.]
第290行预测完成！
[86.]
第291行预测完成！
[97.]
第292行预测完成！
[145.]
第293行预测完成！
[321.]
第294行预测完成！
[937.]
第295行预测完成！
[1271.]
第296行预测完成！
[1057.]
第297行预测完成！
[956.]
第298行预测完成！
[1554.]
第299行预测完成！
[2014.]
第300行预测完成！
[1389.]
第301行预测完成！
[1012.]
第302行预测完成！
[1042.]
第303行预测完成！
[985.]
第304行预测完成！
[1569.]
第305行预测完成！
[1570.]
第306行预测完成！
[1126.]
第307行预测完成！
[967.]
第308行预测完成！
[923.]
第309行预测完成！
[844.]
第310行预测完成！
[359.]
第311行预测完成！
[146.]
第312行预测完成！
[113.]
第313行预测完成！
[111.]
第314行预测完成！
[114.]
第315行预测完成！
[124.]
第316行预测完成！
[155.]
第317行预测完成！
[298.]
第318行预测完成！
[660.]
第319行预测完成！
[945.]
第320行预测完成！
[900.]
第321行预测完成！
[879.]
第322行预测完成！
[1392.]
第323行预测完成！
[1718.]
第324行预测完成！
[1188.]
第325行预测完成！
[940.]
第326行预测完成！
[871.]
第327行预测完成！
[895.]
第328行预测完成！
[1365.]
第329行预测完成！
[1403.]
第330行预测完成！
[1021.]
第331行预测完成！
[872.]
第332行预测完成！
[807.]
第333行预测完成！
[782.]
第334行预测完成！


第192行预测完成！
[50.]
第193行预测完成！
[32.]
第194行预测完成！
[28.]
第195行预测完成！
[24.]
第196行预测完成！
[31.]
第197行预测完成！
[92.]
第198行预测完成！
[413.]
第199行预测完成！
[734.]
第200行预测完成！
[541.]
第201行预测完成！
[381.]
第202行预测完成！
[794.]
第203行预测完成！
[1099.]
第204行预测完成！
[734.]
第205行预测完成！
[588.]
第206行预测完成！
[489.]
第207行预测完成！
[429.]
第208行预测完成！
[899.]
第209行预测完成！
[923.]
第210行预测完成！
[620.]
第211行预测完成！
[541.]
第212行预测完成！
[589.]
第213行预测完成！
[590.]
第214行预测完成！
[357.]
第215行预测完成！
[115.]
第216行预测完成！
[47.]
第217行预测完成！
[30.]
第218行预测完成！
[23.]
第219行预测完成！
[19.]
第220行预测完成！
[30.]
第221行预测完成！
[97.]
第222行预测完成！
[475.]
第223行预测完成！
[733.]
第224行预测完成！
[553.]
第225行预测完成！
[379.]
第226行预测完成！
[776.]
第227行预测完成！
[1086.]
第228行预测完成！
[748.]
第229行预测完成！
[592.]
第230行预测完成！
[480.]
第231行预测完成！
[424.]
第232行预测完成！
[896.]
第233行预测完成！
[948.]
第234行预测完成！
[603.]
第235行预测完成！
[547.]
第236行预测完成！
[606.]
第237行预测完成！
[609.]
第238行预测完成！
[370.]
第239行预测完成！
[115.]
第240行预测完成！
[44.]
第241行预测完成！
[25.]
第242行预测完成！
[19.]
第243行预测完成！
[17.]
第244行预测完成！
[29.]
第245行预测完成！
[102.]
第246行预测完成！
[490.]
第247行预测完成！
[757.]
第248行预测

第104行预测完成！
[544.]
第105行预测完成！
[461.]
第106行预测完成！
[728.]
第107行预测完成！
[934.]
第108行预测完成！
[663.]
第109行预测完成！
[521.]
第110行预测完成！
[469.]
第111行预测完成！
[492.]
第112行预测完成！
[753.]
第113行预测完成！
[693.]
第114行预测完成！
[487.]
第115行预测完成！
[432.]
第116行预测完成！
[442.]
第117行预测完成！
[458.]
第118行预测完成！
[282.]
第119行预测完成！
[169.]
第120行预测完成！
[143.]
第121行预测完成！
[125.]
第122行预测完成！
[115.]
第123行预测完成！
[107.]
第124行预测完成！
[115.]
第125行预测完成！
[162.]
第126行预测完成！
[342.]
第127行预测完成！
[465.]
第128行预测完成！
[472.]
第129行预测完成！
[445.]
第130行预测完成！
[630.]
第131行预测完成！
[779.]
第132行预测完成！
[574.]
第133行预测完成！
[471.]
第134行预测完成！
[440.]
第135行预测完成！
[449.]
第136行预测完成！
[595.]
第137行预测完成！
[597.]
第138行预测完成！
[461.]
第139行预测完成！
[404.]
第140行预测完成！
[391.]
第141行预测完成！
[402.]
第142行预测完成！
[262.]
第143行预测完成！
[180.]
第144行预测完成！
[155.]
第145行预测完成！
[138.]
第146行预测完成！
[131.]
第147行预测完成！
[124.]
第148行预测完成！
[128.]
第149行预测完成！
[168.]
第150行预测完成！
[293.]
第151行预测完成！
[395.]
第152行预测完成！
[442.]
第153行预测完成！
[440.]
第154行预测完成！
[625.]
第155行预测完成！
[753.]
第156行预测完成！
[530.]
第157行预测完成！
[447.]
第158行预测完成！
[415.]
第159行预测完成！

第5行预测完成！
[122.]
第6行预测完成！
[401.]
第7行预测完成！
[604.]
第8行预测完成！
[628.]
第9行预测完成！
[493.]
第10行预测完成！
[930.]
第11行预测完成！
[1288.]
第12行预测完成！
[933.]
第13行预测完成！
[695.]
第14行预测完成！
[587.]
第15行预测完成！
[542.]
第16行预测完成！
[1005.]
第17行预测完成！
[1174.]
第18行预测完成！
[863.]
第19行预测完成！
[732.]
第20行预测完成！
[745.]
第21行预测完成！
[787.]
第22行预测完成！
[431.]
第23行预测完成！
[153.]
第24行预测完成！
[97.]
第25行预测完成！
[72.]
第26行预测完成！
[62.]
第27行预测完成！
[56.]
第28行预测完成！
[62.]
第29行预测完成！
[128.]
第30行预测完成！
[600.]
第31行预测完成！
[799.]
第32行预测完成！
[732.]
第33行预测完成！
[518.]
第34行预测完成！
[986.]
第35行预测完成！
[1348.]
第36行预测完成！
[933.]
第37行预测完成！
[723.]
第38行预测完成！
[626.]
第39行预测完成！
[545.]
第40行预测完成！
[1097.]
第41行预测完成！
[1206.]
第42行预测完成！
[839.]
第43行预测完成！
[735.]
第44行预测完成！
[770.]
第45行预测完成！
[815.]
第46行预测完成！
[406.]
第47行预测完成！
[150.]
第48行预测完成！
[94.]
第49行预测完成！
[67.]
第50行预测完成！
[58.]
第51行预测完成！
[55.]
第52行预测完成！
[65.]
第53行预测完成！
[146.]
第54行预测完成！
[621.]
第55行预测完成！
[857.]
第56行预测完成！
[739.]
第57行预测完成！
[518.]
第58行预测完成！
[987.]
第59行预测完成！
[1420.]
第60行预测完成！
[937.]
第61行预测完成！
[717.]
第62行预测完成！
[598.]
第63行预测完成！
[556.]
第64行预

第477行预测完成！
[585.]
第478行预测完成！
[327.]
第479行预测完成！
[109.]
第480行预测完成！
[55.]
第481行预测完成！
[39.]
第482行预测完成！
[31.]
第483行预测完成！
[30.]
第484行预测完成！
[39.]
第485行预测完成！
[106.]
第486行预测完成！
[307.]
第487行预测完成！
[521.]
第488行预测完成！
[583.]
第489行预测完成！
[539.]
第490行预测完成！
[864.]
第491行预测完成！
[1007.]
第492行预测完成！
[730.]
第493行预测完成！
[604.]
第494行预测完成！
[562.]
第495行预测完成！
[573.]
第496行预测完成！
[805.]
第497行预测完成！
[838.]
第498行预测完成！
[658.]
第499行预测完成！
[564.]
第500行预测完成！
[552.]
第501行预测完成！
[577.]
第502行预测完成！
[337.]
第503行预测完成！
[126.]
第504行预测完成！
[64.]
第505行预测完成！
[46.]
第506行预测完成！
[42.]
第507行预测完成！
[40.]
第508行预测完成！
[49.]
第509行预测完成！
[118.]
第510行预测完成！
[561.]
第511行预测完成！
[891.]
第512行预测完成！
[742.]
第513行预测完成！
[577.]
第514行预测完成！
[1059.]
第515行预测完成！
[1359.]
第516行预测完成！
[931.]
第517行预测完成！
[779.]
第518行预测完成！
[686.]
第519行预测完成！
[642.]
第520行预测完成！
[1222.]
第521行预测完成！
[1209.]
第522行预测完成！
[813.]
第523行预测完成！
[752.]
第524行预测完成！
[803.]
第525行预测完成！
[817.]
第526行预测完成！
[389.]
第527行预测完成！
[108.]
第528行预测完成！
[58.]
第529行预测完成！
[46.]
第530行预测完成！
[39.]
第531行预测完成！
[36.]
第532行预测完成！
[45.]
第5

第394行预测完成！
[389.]
第395行预测完成！
[485.]
第396行预测完成！
[360.]
第397行预测完成！
[307.]
第398行预测完成！
[277.]
第399行预测完成！
[250.]
第400行预测完成！
[418.]
第401行预测完成！
[373.]
第402行预测完成！
[275.]
第403行预测完成！
[237.]
第404行预测完成！
[222.]
第405行预测完成！
[182.]
第406行预测完成！
[68.]
第407行预测完成！
[17.]
第408行预测完成！
[11.]
第409行预测完成！
[8.]
第410行预测完成！
[8.]
第411行预测完成！
[8.]
第412行预测完成！
[16.]
第413行预测完成！
[50.]
第414行预测完成！
[223.]
第415行预测完成！
[470.]
第416行预测完成！
[378.]
第417行预测完成！
[285.]
第418行预测完成！
[392.]
第419行预测完成！
[471.]
第420行预测完成！
[360.]
第421行预测完成！
[309.]
第422行预测完成！
[285.]
第423行预测完成！
[251.]
第424行预测完成！
[426.]
第425行预测完成！
[371.]
第426行预测完成！
[275.]
第427行预测完成！
[230.]
第428行预测完成！
[207.]
第429行预测完成！
[174.]
第430行预测完成！
[59.]
第431行预测完成！
[16.]
第432行预测完成！
[9.]
第433行预测完成！
[6.]
第434行预测完成！
[6.]
第435行预测完成！
[5.]
第436行预测完成！
[10.]
第437行预测完成！
[41.]
第438行预测完成！
[214.]
第439行预测完成！
[454.]
第440行预测完成！
[365.]
第441行预测完成！
[272.]
第442行预测完成！
[379.]
第443行预测完成！
[445.]
第444行预测完成！
[345.]
第445行预测完成！
[298.]
第446行预测完成！
[278.]
第447行预测完成！
[248.]
第448行预测完成！
[401.]
第449行预测完成！
[336.]
第450行预测完成！
[261

第306行预测完成！
[3878.]
第307行预测完成！
[2892.]
第308行预测完成！
[2655.]
第309行预测完成！
[2062.]
第310行预测完成！
[917.]
第311行预测完成！
[384.]
第312行预测完成！
[292.]
第313行预测完成！
[447.]
第314行预测完成！
[514.]
第315行预测完成！
[570.]
第316行预测完成！
[620.]
第317行预测完成！
[725.]
第318行预测完成！
[1643.]
第319行预测完成！
[2488.]
第320行预测完成！
[2746.]
第321行预测完成！
[2292.]
第322行预测完成！
[4771.]
第323行预测完成！
[5980.]
第324行预测完成！
[4541.]
第325行预测完成！
[2562.]
第326行预测完成！
[2424.]
第327行预测完成！
[2432.]
第328行预测完成！
[4714.]
第329行预测完成！
[4397.]
第330行预测完成！
[3925.]
第331行预测完成！
[2937.]
第332行预测完成！
[2575.]
第333行预测完成！
[1877.]
第334行预测完成！
[843.]
第335行预测完成！
[372.]
第336行预测完成！
[294.]
第337行预测完成！
[454.]
第338行预测完成！
[562.]
第339行预测完成！
[657.]
第340行预测完成！
[745.]
第341行预测完成！
[884.]
第342行预测完成！
[3050.]
第343行预测完成！
[3458.]
第344行预测完成！
[3878.]
第345行预测完成！
[2459.]
第346行预测完成！
[5104.]
第347行预测完成！
[6656.]
第348行预测完成！
[5092.]
第349行预测完成！
[2598.]
第350行预测完成！
[2677.]
第351行预测完成！
[2697.]
第352行预测完成！
[5082.]
第353行预测完成！
[4677.]
第354行预测完成！
[4335.]
第355行预测完成！
[3566.]
第356行预测完成！
[2811.]
第357行预测完成！
[2352.]
第358行预测完成！
[1003.]
第359行预测完成

第212行预测完成！
[204.]
第213行预测完成！
[169.]
第214行预测完成！
[84.]
第215行预测完成！
[40.]
第216行预测完成！
[23.]
第217行预测完成！
[18.]
第218行预测完成！
[18.]
第219行预测完成！
[19.]
第220行预测完成！
[28.]
第221行预测完成！
[58.]
第222行预测完成！
[183.]
第223行预测完成！
[359.]
第224行预测完成！
[282.]
第225行预测完成！
[222.]
第226行预测完成！
[303.]
第227行预测完成！
[391.]
第228行预测完成！
[288.]
第229行预测完成！
[241.]
第230行预测完成！
[206.]
第231行预测完成！
[198.]
第232行预测完成！
[308.]
第233行预测完成！
[289.]
第234行预测完成！
[235.]
第235行预测完成！
[210.]
第236行预测完成！
[203.]
第237行预测完成！
[164.]
第238行预测完成！
[81.]
第239行预测完成！
[42.]
第240行预测完成！
[22.]
第241行预测完成！
[16.]
第242行预测完成！
[15.]
第243行预测完成！
[14.]
第244行预测完成！
[19.]
第245行预测完成！
[51.]
第246行预测完成！
[188.]
第247行预测完成！
[371.]
第248行预测完成！
[281.]
第249行预测完成！
[222.]
第250行预测完成！
[300.]
第251行预测完成！
[386.]
第252行预测完成！
[293.]
第253行预测完成！
[246.]
第254行预测完成！
[211.]
第255行预测完成！
[199.]
第256行预测完成！
[316.]
第257行预测完成！
[277.]
第258行预测完成！
[221.]
第259行预测完成！
[196.]
第260行预测完成！
[185.]
第261行预测完成！
[155.]
第262行预测完成！
[84.]
第263行预测完成！
[36.]
第264行预测完成！
[18.]
第265行预测完成！
[11.]
第266行预测完成！
[9.]
第267行预测完成！
[8.]
第268行预测完成！
[14.]

第127行预测完成！
[1652.]
第128行预测完成！
[935.]
第129行预测完成！
[732.]
第130行预测完成！
[2523.]
第131行预测完成！
[3140.]
第132行预测完成！
[1420.]
第133行预测完成！
[851.]
第134行预测完成！
[786.]
第135行预测完成！
[969.]
第136行预测完成！
[2541.]
第137行预测完成！
[2089.]
第138行预测完成！
[1223.]
第139行预测完成！
[871.]
第140行预测完成！
[620.]
第141行预测完成！
[313.]
第142行预测完成！
[112.]
第143行预测完成！
[25.]
第144行预测完成！
[6.]
第145行预测完成！
[2.]
第146行预测完成！
[1.]
第147行预测完成！
[4.]
第148行预测完成！
[21.]
第149行预测完成！
[131.]
第150行预测完成！
[527.]
第151行预测完成！
[996.]
第152行预测完成！
[920.]
第153行预测完成！
[796.]
第154行预测完成！
[2327.]
第155行预测完成！
[2719.]
第156行预测完成！
[1393.]
第157行预测完成！
[855.]
第158行预测完成！
[825.]
第159行预测完成！
[1029.]
第160行预测完成！
[2530.]
第161行预测完成！
[2118.]
第162行预测完成！
[1324.]
第163行预测完成！
[870.]
第164行预测完成！
[601.]
第165行预测完成！
[310.]
第166行预测完成！
[110.]
第167行预测完成！
[30.]
第168行预测完成！
[15.]
第169行预测完成！
[13.]
第170行预测完成！
[13.]
第171行预测完成！
[14.]
第172行预测完成！
[37.]
第173行预测完成！
[170.]
第174行预测完成！
[1126.]
第175行预测完成！
[1527.]
第176行预测完成！
[1001.]
第177行预测完成！
[872.]
第178行预测完成！
[3139.]
第179行预测完成！
[3633.]
第180行预测完成！
[1447.]
第181行预测完成！
[887.]
第182行预

[1023.]
第36行预测完成！
[893.]
第37行预测完成！
[627.]
第38行预测完成！
[604.]
第39行预测完成！
[506.]
第40行预测完成！
[1040.]
第41行预测完成！
[991.]
第42行预测完成！
[695.]
第43行预测完成！
[601.]
第44行预测完成！
[602.]
第45行预测完成！
[595.]
第46行预测完成！
[391.]
第47行预测完成！
[228.]
第48行预测完成！
[179.]
第49行预测完成！
[153.]
第50行预测完成！
[142.]
第51行预测完成！
[134.]
第52行预测完成！
[136.]
第53行预测完成！
[205.]
第54行预测完成！
[608.]
第55行预测完成！
[621.]
第56行预测完成！
[620.]
第57行预测完成！
[498.]
第58行预测完成！
[847.]
第59行预测完成！
[1091.]
第60行预测完成！
[865.]
第61行预测完成！
[616.]
第62行预测完成！
[590.]
第63行预测完成！
[520.]
第64行预测完成！
[1004.]
第65行预测完成！
[1048.]
第66行预测完成！
[733.]
第67行预测完成！
[637.]
第68行预测完成！
[666.]
第69行预测完成！
[613.]
第70行预测完成！
[395.]
第71行预测完成！
[197.]
第72行预测完成！
[176.]
第73行预测完成！
[158.]
第74行预测完成！
[144.]
第75行预测完成！
[139.]
第76行预测完成！
[140.]
第77行预测完成！
[212.]
第78行预测完成！
[605.]
第79行预测完成！
[653.]
第80行预测完成！
[603.]
第81行预测完成！
[490.]
第82行预测完成！
[846.]
第83行预测完成！
[1122.]
第84行预测完成！
[852.]
第85行预测完成！
[625.]
第86行预测完成！
[560.]
第87行预测完成！
[506.]
第88行预测完成！
[959.]
第89行预测完成！
[1001.]
第90行预测完成！
[691.]
第91行预测完成！
[619.]
第92行预测完成！
[605.]
第93行预测完成！
[592.]


第499行预测完成！
[462.]
第500行预测完成！
[456.]
第501行预测完成！
[485.]
第502行预测完成！
[323.]
第503行预测完成！
[190.]
第504行预测完成！
[164.]
第505行预测完成！
[152.]
第506行预测完成！
[146.]
第507行预测完成！
[142.]
第508行预测完成！
[150.]
第509行预测完成！
[227.]
第510行预测完成！
[481.]
第511行预测完成！
[696.]
第512行预测完成！
[624.]
第513行预测完成！
[531.]
第514行预测完成！
[865.]
第515行预测完成！
[1074.]
第516行预测完成！
[849.]
第517行预测完成！
[665.]
第518行预测完成！
[603.]
第519行预测完成！
[565.]
第520行预测完成！
[999.]
第521行预测完成！
[1001.]
第522行预测完成！
[701.]
第523行预测完成！
[625.]
第524行预测完成！
[644.]
第525行预测完成！
[619.]
第526行预测完成！
[383.]
第527行预测完成！
[191.]
第528行预测完成！
[164.]
第529行预测完成！
[152.]
第530行预测完成！
[146.]
第531行预测完成！
[139.]
第532行预测完成！
[140.]
第533行预测完成！
[221.]
第534行预测完成！
[592.]
第535行预测完成！
[741.]
第536行预测完成！
[631.]
第537行预测完成！
[542.]
第538行预测完成！
[942.]
第539行预测完成！
[1126.]
第540行预测完成！
[853.]
第541行预测完成！
[673.]
第542行预测完成！
[620.]
第543行预测完成！
[570.]
第544行预测完成！
[1044.]
第545行预测完成！
[1000.]
第546行预测完成！
[694.]
第547行预测完成！
[645.]
第548行预测完成！
[657.]
第549行预测完成！
[630.]
第550行预测完成！
[367.]
第551行预测完成！
******删减前地点数据大小：1272******
******删减后地点数据大小：552**

第394行预测完成！
[6964.]
第395行预测完成！
[8351.]
第396行预测完成！
[5949.]
第397行预测完成！
[2807.]
第398行预测完成！
[2628.]
第399行预测完成！
[2744.]
第400行预测完成！
[6586.]
第401行预测完成！
[6109.]
第402行预测完成！
[4703.]
第403行预测完成！
[3368.]
第404行预测完成！
[2715.]
第405行预测完成！
[2374.]
第406行预测完成！
[1110.]
第407行预测完成！
[617.]
第408行预测完成！
[492.]
第409行预测完成！
[600.]
第410行预测完成！
[822.]
第411行预测完成！
[849.]
第412行预测完成！
[850.]
第413行预测完成！
[968.]
第414行预测完成！
[3428.]
第415行预测完成！
[4139.]
第416行预测完成！
[4182.]
第417行预测完成！
[2736.]
第418行预测完成！
[7095.]
第419行预测完成！
[8337.]
第420行预测完成！
[6013.]
第421行预测完成！
[2786.]
第422行预测完成！
[2755.]
第423行预测完成！
[2895.]
第424行预测完成！
[6645.]
第425行预测完成！
[6546.]
第426行预测完成！
[4953.]
第427行预测完成！
[3638.]
第428行预测完成！
[2665.]
第429行预测完成！
[2269.]
第430行预测完成！
[1184.]
第431行预测完成！
[694.]
第432行预测完成！
[550.]
第433行预测完成！
[699.]
第434行预测完成！
[925.]
第435行预测完成！
[846.]
第436行预测完成！
[908.]
第437行预测完成！
[944.]
第438行预测完成！
[3024.]
第439行预测完成！
[3689.]
第440行预测完成！
[3834.]
第441行预测完成！
[2807.]
第442行预测完成！
[7130.]
第443行预测完成！
[8269.]
第444行预测完成！
[5868.]
第445行预测完成！
[2744.]
第446行预测完成！
[2774.]
第447行预测

第301行预测完成！
[307.]
第302行预测完成！
[290.]
第303行预测完成！
[258.]
第304行预测完成！
[389.]
第305行预测完成！
[382.]
第306行预测完成！
[304.]
第307行预测完成！
[275.]
第308行预测完成！
[258.]
第309行预测完成！
[228.]
第310行预测完成！
[96.]
第311行预测完成！
[30.]
第312行预测完成！
[12.]
第313行预测完成！
[6.]
第314行预测完成！
[4.]
第315行预测完成！
[3.]
第316行预测完成！
[5.]
第317行预测完成！
[30.]
第318行预测完成！
[137.]
第319行预测完成！
[267.]
第320行预测完成！
[321.]
第321行预测完成！
[272.]
第322行预测完成！
[379.]
第323行预测完成！
[439.]
第324行预测完成！
[347.]
第325行预测完成！
[288.]
第326行预测完成！
[264.]
第327行预测完成！
[249.]
第328行预测完成！
[363.]
第329行预测完成！
[362.]
第330行预测完成！
[293.]
第331行预测完成！
[262.]
第332行预测完成！
[252.]
第333行预测完成！
[216.]
第334行预测完成！
[96.]
第335行预测完成！
[32.]
第336行预测完成！
[19.]
第337行预测完成！
[16.]
第338行预测完成！
[17.]
第339行预测完成！
[18.]
第340行预测完成！
[28.]
第341行预测完成！
[58.]
第342行预测完成！
[246.]
第343行预测完成！
[512.]
第344行预测完成！
[418.]
第345行预测完成！
[330.]
第346行预测完成！
[453.]
第347行预测完成！
[597.]
第348行预测完成！
[428.]
第349行预测完成！
[359.]
第350行预测完成！
[318.]
第351行预测完成！
[293.]
第352行预测完成！
[497.]
第353行预测完成！
[506.]
第354行预测完成！
[369.]
第355行预测完成！
[323.]
第356行预测完成！
[330.]
第357行预测完成！
[

第221行预测完成！
[66.]
第222行预测完成！
[353.]
第223行预测完成！
[617.]
第224行预测完成！
[618.]
第225行预测完成！
[511.]
第226行预测完成！
[609.]
第227行预测完成！
[603.]
第228行预测完成！
[516.]
第229行预测完成！
[512.]
第230行预测完成！
[504.]
第231行预测完成！
[461.]
第232行预测完成！
[646.]
第233行预测完成！
[587.]
第234行预测完成！
[408.]
第235行预测完成！
[372.]
第236行预测完成！
[307.]
第237行预测完成！
[229.]
第238行预测完成！
[55.]
第239行预测完成！
[9.]
第240行预测完成！
[7.]
第241行预测完成！
[6.]
第242行预测完成！
[7.]
第243行预测完成！
[12.]
第244行预测完成！
[24.]
第245行预测完成！
[69.]
第246行预测完成！
[337.]
第247行预测完成！
[615.]
第248行预测完成！
[649.]
第249行预测完成！
[542.]
第250行预测完成！
[666.]
第251行预测完成！
[630.]
第252行预测完成！
[525.]
第253行预测完成！
[526.]
第254行预测完成！
[536.]
第255行预测完成！
[478.]
第256行预测完成！
[652.]
第257行预测完成！
[552.]
第258行预测完成！
[383.]
第259行预测完成！
[322.]
第260行预测完成！
[274.]
第261行预测完成！
[205.]
第262行预测完成！
[52.]
第263行预测完成！
[8.]
第264行预测完成！
[6.]
第265行预测完成！
[3.]
第266行预测完成！
[4.]
第267行预测完成！
[4.]
第268行预测完成！
[12.]
第269行预测完成！
[51.]
第270行预测完成！
[287.]
第271行预测完成！
[547.]
第272行预测完成！
[577.]
第273行预测完成！
[472.]
第274行预测完成！
[584.]
第275行预测完成！
[601.]
第276行预测完成！
[497.]
第277行预测完成！
[494.]


第142行预测完成！
[140.]
第143行预测完成！
[30.]
第144行预测完成！
[11.]
第145行预测完成！
[6.]
第146行预测完成！
[5.]
第147行预测完成！
[5.]
第148行预测完成！
[13.]
第149行预测完成！
[66.]
第150行预测完成！
[261.]
第151行预测完成！
[464.]
第152行预测完成！
[556.]
第153行预测完成！
[586.]
第154行预测完成！
[751.]
第155行预测完成！
[741.]
第156行预测完成！
[556.]
第157行预测完成！
[486.]
第158行预测完成！
[525.]
第159行预测完成！
[601.]
第160行预测完成！
[781.]
第161行预测完成！
[700.]
第162行预测完成！
[505.]
第163行预测完成！
[431.]
第164行预测完成！
[401.]
第165行预测完成！
[364.]
第166行预测完成！
[142.]
第167行预测完成！
[35.]
第168行预测完成！
[17.]
第169行预测完成！
[14.]
第170行预测完成！
[15.]
第171行预测完成！
[18.]
第172行预测完成！
[31.]
第173行预测完成！
[105.]
第174行预测完成！
[422.]
第175行预测完成！
[665.]
第176行预测完成！
[687.]
第177行预测完成！
[645.]
第178行预测完成！
[870.]
第179行预测完成！
[896.]
第180行预测完成！
[708.]
第181行预测完成！
[627.]
第182行预测完成！
[663.]
第183行预测完成！
[702.]
第184行预测完成！
[1028.]
第185行预测完成！
[870.]
第186行预测完成！
[627.]
第187行预测完成！
[607.]
第188行预测完成！
[588.]
第189行预测完成！
[498.]
第190行预测完成！
[166.]
第191行预测完成！
[36.]
第192行预测完成！
[14.]
第193行预测完成！
[8.]
第194行预测完成！
[8.]
第195行预测完成！
[10.]
第196行预测完成！
[24.]
第197行预测完成！
[96.]
第198行预测完成！
[510.

第58行预测完成！
[1069.]
第59行预测完成！
[855.]
第60行预测完成！
[749.]
第61行预测完成！
[776.]
第62行预测完成！
[850.]
第63行预测完成！
[873.]
第64行预测完成！
[1318.]
第65行预测完成！
[886.]
第66行预测完成！
[561.]
第67行预测完成！
[502.]
第68行预测完成！
[431.]
第69行预测完成！
[301.]
第70行预测完成！
[104.]
第71行预测完成！
[30.]
第72行预测完成！
[13.]
第73行预测完成！
[8.]
第74行预测完成！
[10.]
第75行预测完成！
[13.]
第76行预测完成！
[27.]
第77行预测完成！
[75.]
第78行预测完成！
[467.]
第79行预测完成！
[768.]
第80行预测完成！
[759.]
第81行预测完成！
[778.]
第82行预测完成！
[1075.]
第83行预测完成！
[927.]
第84行预测完成！
[744.]
第85行预测完成！
[786.]
第86行预测完成！
[825.]
第87行预测完成！
[861.]
第88行预测完成！
[1242.]
第89行预测完成！
[843.]
第90行预测完成！
[521.]
第91行预测完成！
[466.]
第92行预测完成！
[404.]
第93行预测完成！
[278.]
第94行预测完成！
[85.]
第95行预测完成！
[17.]
第96行预测完成！
[9.]
第97行预测完成！
[7.]
第98行预测完成！
[7.]
第99行预测完成！
[9.]
第100行预测完成！
[20.]
第101行预测完成！
[77.]
第102行预测完成！
[459.]
第103行预测完成！
[794.]
第104行预测完成！
[805.]
第105行预测完成！
[755.]
第106行预测完成！
[1006.]
第107行预测完成！
[852.]
第108行预测完成！
[723.]
第109行预测完成！
[780.]
第110行预测完成！
[814.]
第111行预测完成！
[816.]
第112行预测完成！
[1174.]
第113行预测完成！
[818.]
第114行预测完成！
[496.]
第115行预测完成！
[416.]
第116行预测完成！
[

第531行预测完成！
[10.]
第532行预测完成！
[24.]
第533行预测完成！
[77.]
第534行预测完成！
[486.]
第535行预测完成！
[861.]
第536行预测完成！
[759.]
第537行预测完成！
[727.]
第538行预测完成！
[1101.]
第539行预测完成！
[990.]
第540行预测完成！
[768.]
第541行预测完成！
[779.]
第542行预测完成！
[804.]
第543行预测完成！
[865.]
第544行预测完成！
[1322.]
第545行预测完成！
[1005.]
第546行预测完成！
[619.]
第547行预测完成！
[574.]
第548行预测完成！
[504.]
第549行预测完成！
[375.]
第550行预测完成！
[126.]
第551行预测完成！
******删减前地点数据大小：1272******
******删减后地点数据大小：552******
[211.]
第0行预测完成！
[133.]
第1行预测完成！
[87.]
第2行预测完成！
[84.]
第3行预测完成！
[80.]
第4行预测完成！
[97.]
第5行预测完成！
[165.]
第6行预测完成！
[494.]
第7行预测完成！
[759.]
第8行预测完成！
[829.]
第9行预测完成！
[671.]
第10行预测完成！
[1130.]
第11行预测完成！
[1417.]
第12行预测完成！
[981.]
第13行预测完成！
[882.]
第14行预测完成！
[760.]
第15行预测完成！
[741.]
第16行预测完成！
[1243.]
第17行预测完成！
[1302.]
第18行预测完成！
[841.]
第19行预测完成！
[781.]
第20行预测完成！
[836.]
第21行预测完成！
[875.]
第22行预测完成！
[505.]
第23行预测完成！
[216.]
第24行预测完成！
[130.]
第25行预测完成！
[112.]
第26行预测完成！
[100.]
第27行预测完成！
[91.]
第28行预测完成！
[93.]
第29行预测完成！
[179.]
第30行预测完成！
[659.]
第31行预测完成！
[979.]
第32行预测完成！
[884.]
第33行预测完成！
[631.]
第34

第437行预测完成！
[211.]
第438行预测完成！
[777.]
第439行预测完成！
[920.]
第440行预测完成！
[864.]
第441行预测完成！
[678.]
第442行预测完成！
[1161.]
第443行预测完成！
[1436.]
第444行预测完成！
[937.]
第445行预测完成！
[797.]
第446行预测完成！
[740.]
第447行预测完成！
[764.]
第448行预测完成！
[1137.]
第449行预测完成！
[1097.]
第450行预测完成！
[745.]
第451行预测完成！
[705.]
第452行预测完成！
[758.]
第453行预测完成！
[787.]
第454行预测完成！
[417.]
第455行预测完成！
[177.]
第456行预测完成！
[144.]
第457行预测完成！
[124.]
第458行预测完成！
[115.]
第459行预测完成！
[109.]
第460行预测完成！
[117.]
第461行预测完成！
[194.]
第462行预测完成！
[445.]
第463行预测完成！
[700.]
第464行预测完成！
[784.]
第465行预测完成！
[673.]
第466行预测完成！
[1028.]
第467行预测完成！
[1260.]
第468行预测完成！
[833.]
第469行预测完成！
[725.]
第470行预测完成！
[690.]
第471行预测完成！
[726.]
第472行预测完成！
[954.]
第473行预测完成！
[953.]
第474行预测完成！
[676.]
第475行预测完成！
[643.]
第476行预测完成！
[652.]
第477行预测完成！
[693.]
第478行预测完成！
[412.]
第479行预测完成！
[196.]
第480行预测完成！
[154.]
第481行预测完成！
[128.]
第482行预测完成！
[120.]
第483行预测完成！
[114.]
第484行预测完成！
[117.]
第485行预测完成！
[208.]
第486行预测完成！
[391.]
第487行预测完成！
[613.]
第488行预测完成！
[711.]
第489行预测完成！
[681.]
第490行预测完成！
[971.]
第491行预测完成！
[1160.]
第49

第345行预测完成！
[444.]
第346行预测完成！
[594.]
第347行预测完成！
[751.]
第348行预测完成！
[578.]
第349行预测完成！
[496.]
第350行预测完成！
[437.]
第351行预测完成！
[444.]
第352行预测完成！
[675.]
第353行预测完成！
[602.]
第354行预测完成！
[448.]
第355行预测完成！
[411.]
第356行预测完成！
[445.]
第357行预测完成！
[511.]
第358行预测完成！
[402.]
第359行预测完成！
[285.]
第360行预测完成！
[272.]
第361行预测完成！
[260.]
第362行预测完成！
[265.]
第363行预测完成！
[267.]
第364行预测完成！
[263.]
第365行预测完成！
[297.]
第366行预测完成！
[537.]
第367行预测完成！
[733.]
第368行预测完成！
[557.]
第369行预测完成！
[455.]
第370行预测完成！
[596.]
第371行预测完成！
[756.]
第372行预测完成！
[582.]
第373行预测完成！
[488.]
第374行预测完成！
[430.]
第375行预测完成！
[440.]
第376行预测完成！
[702.]
第377行预测完成！
[629.]
第378行预测完成！
[456.]
第379行预测完成！
[417.]
第380行预测完成！
[448.]
第381行预测完成！
[505.]
第382行预测完成！
[408.]
第383行预测完成！
[304.]
第384行预测完成！
[290.]
第385行预测完成！
[286.]
第386行预测完成！
[276.]
第387行预测完成！
[270.]
第388行预测完成！
[258.]
第389行预测完成！
[301.]
第390行预测完成！
[522.]
第391行预测完成！
[736.]
第392行预测完成！
[563.]
第393行预测完成！
[452.]
第394行预测完成！
[592.]
第395行预测完成！
[762.]
第396行预测完成！
[576.]
第397行预测完成！
[488.]
第398行预测完成！
[420.]
第399行预测完成！
[415.]
第400行预测完成！

第257行预测完成！
[869.]
第258行预测完成！
[568.]
第259行预测完成！
[476.]
第260行预测完成！
[491.]
第261行预测完成！
[572.]
第262行预测完成！
[310.]
第263行预测完成！
[157.]
第264行预测完成！
[126.]
第265行预测完成！
[113.]
第266行预测完成！
[104.]
第267行预测完成！
[100.]
第268行预测完成！
[108.]
第269行预测完成！
[167.]
第270行预测完成！
[554.]
第271行预测完成！
[590.]
第272行预测完成！
[483.]
第273行预测完成！
[384.]
第274行预测完成！
[695.]
第275行预测完成！
[973.]
第276行预测完成！
[682.]
第277行预测完成！
[515.]
第278行预测完成！
[465.]
第279行预测完成！
[423.]
第280行预测完成！
[831.]
第281行预测完成！
[800.]
第282行预测完成！
[519.]
第283行预测完成！
[446.]
第284行预测完成！
[464.]
第285行预测完成！
[521.]
第286行预测完成！
[289.]
第287行预测完成！
[161.]
第288行预测完成！
[126.]
第289行预测完成！
[109.]
第290行预测完成！
[100.]
第291行预测完成！
[95.]
第292行预测完成！
[101.]
第293行预测完成！
[151.]
第294行预测完成！
[385.]
第295行预测完成！
[482.]
第296行预测完成！
[449.]
第297行预测完成！
[411.]
第298行预测完成！
[677.]
第299行预测完成！
[877.]
第300行预测完成！
[605.]
第301行预测完成！
[471.]
第302行预测完成！
[433.]
第303行预测完成！
[412.]
第304行预测完成！
[717.]
第305行预测完成！
[716.]
第306行预测完成！
[485.]
第307行预测完成！
[415.]
第308行预测完成！
[415.]
第309行预测完成！
[476.]
第310行预测完成！
[267.]
第311行预测完成！
[162.]
第312行预测完成！


第163行预测完成！
[662.]
第164行预测完成！
[623.]
第165行预测完成！
[557.]
第166行预测完成！
[217.]
第167行预测完成！
[48.]
第168行预测完成！
[21.]
第169行预测完成！
[16.]
第170行预测完成！
[16.]
第171行预测完成！
[17.]
第172行预测完成！
[27.]
第173行预测完成！
[70.]
第174行预测完成！
[327.]
第175行预测完成！
[898.]
第176行预测完成！
[1056.]
第177行预测完成！
[1029.]
第178行预测完成！
[1447.]
第179行预测完成！
[1269.]
第180行预测完成！
[1091.]
第181行预测完成！
[1336.]
第182行预测完成！
[1295.]
第183行预测完成！
[1240.]
第184行预测完成！
[1489.]
第185行预测完成！
[1317.]
第186行预测完成！
[1102.]
第187行预测完成！
[974.]
第188行预测完成！
[895.]
第189行预测完成！
[760.]
第190行预测完成！
[307.]
第191行预测完成！
[55.]
第192行预测完成！
[17.]
第193行预测完成！
[10.]
第194行预测完成！
[9.]
第195行预测完成！
[8.]
第196行预测完成！
[20.]
第197行预测完成！
[69.]
第198行预测完成！
[424.]
第199行预测完成！
[1266.]
第200行预测完成！
[1329.]
第201行预测完成！
[1224.]
第202行预测完成！
[1727.]
第203行预测完成！
[1416.]
第204行预测完成！
[1312.]
第205行预测完成！
[1550.]
第206行预测完成！
[1445.]
第207行预测完成！
[1294.]
第208行预测完成！
[1764.]
第209行预测完成！
[1595.]
第210行预测完成！
[1256.]
第211行预测完成！
[992.]
第212行预测完成！
[901.]
第213行预测完成！
[764.]
第214行预测完成！
[292.]
第215行预测完成！
[53.]
第216行预测完成！
[16.]
第217行预测完成！
[10.]
第218行预

第68行预测完成！
[702.]
第69行预测完成！
[724.]
第70行预测完成！
[406.]
第71行预测完成！
[119.]
第72行预测完成！
[65.]
第73行预测完成！
[52.]
第74行预测完成！
[51.]
第75行预测完成！
[50.]
第76行预测完成！
[58.]
第77行预测完成！
[151.]
第78行预测完成！
[598.]
第79行预测完成！
[822.]
第80行预测完成！
[696.]
第81行预测完成！
[501.]
第82行预测完成！
[884.]
第83行预测完成！
[1262.]
第84行预测完成！
[847.]
第85行预测完成！
[721.]
第86行预测完成！
[620.]
第87行预测完成！
[527.]
第88行预测完成！
[1001.]
第89行预测完成！
[1044.]
第90行预测完成！
[656.]
第91行预测完成！
[661.]
第92行预测完成！
[701.]
第93行预测完成！
[700.]
第94行预测完成！
[376.]
第95行预测完成！
[100.]
第96行预测完成！
[64.]
第97行预测完成！
[54.]
第98行预测完成！
[46.]
第99行预测完成！
[47.]
第100行预测完成！
[60.]
第101行预测完成！
[155.]
第102行预测完成！
[540.]
第103行预测完成！
[840.]
第104行预测完成！
[668.]
第105行预测完成！
[491.]
第106行预测完成！
[884.]
第107行预测完成！
[1169.]
第108行预测完成！
[808.]
第109行预测完成！
[654.]
第110行预测完成！
[601.]
第111行预测完成！
[517.]
第112行预测完成！
[913.]
第113行预测完成！
[902.]
第114行预测完成！
[593.]
第115行预测完成！
[618.]
第116行预测完成！
[629.]
第117行预测完成！
[624.]
第118行预测完成！
[353.]
第119行预测完成！
[123.]
第120行预测完成！
[64.]
第121行预测完成！
[49.]
第122行预测完成！
[43.]
第123行预测完成！
[39.]
第124行预测完成！
[45.]
第125行预测完成！
[111.]

第536行预测完成！
[721.]
第537行预测完成！
[541.]
第538行预测完成！
[1010.]
第539行预测完成！
[1307.]
第540行预测完成！
[877.]
第541行预测完成！
[700.]
第542行预测完成！
[630.]
第543行预测完成！
[591.]
第544行预测完成！
[1154.]
第545行预测完成！
[1100.]
第546行预测完成！
[713.]
第547行预测完成！
[689.]
第548行预测完成！
[750.]
第549行预测完成！
[755.]
第550行预测完成！
[361.]
第551行预测完成！
******删减前地点数据大小：1272******
******删减后地点数据大小：552******
[10.]
第0行预测完成！
[9.]
第1行预测完成！
[8.]
第2行预测完成！
[10.]
第3行预测完成！
[11.]
第4行预测完成！
[18.]
第5行预测完成！
[40.]
第6行预测完成！
[163.]
第7行预测完成！
[332.]
第8行预测完成！
[394.]
第9行预测完成！
[291.]
第10行预测完成！
[311.]
第11行预测完成！
[331.]
第12行预测完成！
[315.]
第13行预测完成！
[299.]
第14行预测完成！
[308.]
第15行预测完成！
[262.]
第16行预测完成！
[334.]
第17行预测完成！
[301.]
第18行预测完成！
[241.]
第19行预测完成！
[201.]
第20行预测完成！
[144.]
第21行预测完成！
[82.]
第22行预测完成！
[26.]
第23行预测完成！
[12.]
第24行预测完成！
[9.]
第25行预测完成！
[8.]
第26行预测完成！
[8.]
第27行预测完成！
[9.]
第28行预测完成！
[14.]
第29行预测完成！
[39.]
第30行预测完成！
[216.]
第31行预测完成！
[441.]
第32行预测完成！
[470.]
第33行预测完成！
[326.]
第34行预测完成！
[353.]
第35行预测完成！
[395.]
第36行预测完成！
[357.]
第37行预测完成！
[322.]
第38行预测完成！
[356.]
第39行预测完成！
[299.]
第40行预测完

第456行预测完成！
[4.]
第457行预测完成！
[1.]
第458行预测完成！
[-0.]
第459行预测完成！
[-0.]
第460行预测完成！
[3.]
第461行预测完成！
[14.]
第462行预测完成！
[97.]
第463行预测完成！
[242.]
第464行预测完成！
[335.]
第465行预测完成！
[302.]
第466行预测完成！
[338.]
第467行预测完成！
[345.]
第468行预测完成！
[290.]
第469行预测完成！
[263.]
第470行预测完成！
[253.]
第471行预测完成！
[233.]
第472行预测完成！
[365.]
第473行预测完成！
[327.]
第474行预测完成！
[263.]
第475行预测完成！
[208.]
第476行预测完成！
[166.]
第477行预测完成！
[99.]
第478行预测完成！
[36.]
第479行预测完成！
[10.]
第480行预测完成！
[5.]
第481行预测完成！
[2.]
第482行预测完成！
[-0.]
第483行预测完成！
[-1.]
第484行预测完成！
[-0.]
第485行预测完成！
[6.]
第486行预测完成！
[41.]
第487行预测完成！
[111.]
第488行预测完成！
[189.]
第489行预测完成！
[229.]
第490行预测完成！
[296.]
第491行预测完成！
[315.]
第492行预测完成！
[255.]
第493行预测完成！
[234.]
第494行预测完成！
[208.]
第495行预测完成！
[218.]
第496行预测完成！
[318.]
第497行预测完成！
[303.]
第498行预测完成！
[253.]
第499行预测完成！
[207.]
第500行预测完成！
[168.]
第501行预测完成！
[104.]
第502行预测完成！
[45.]
第503行预测完成！
[19.]
第504行预测完成！
[13.]
第505行预测完成！
[12.]
第506行预测完成！
[13.]
第507行预测完成！
[14.]
第508行预测完成！
[18.]
第509行预测完成！
[36.]
第510行预测完成！
[149.]
第511行预测完成！
[380.]
第512行预测完成！
[383.]
第513行

[468.]
第368行预测完成！
[436.]
第369行预测完成！
[371.]
第370行预测完成！
[600.]
第371行预测完成！
[791.]
第372行预测完成！
[570.]
第373行预测完成！
[427.]
第374行预测完成！
[408.]
第375行预测完成！
[380.]
第376行预测完成！
[719.]
第377行预测完成！
[671.]
第378行预测完成！
[480.]
第379行预测完成！
[404.]
第380行预测完成！
[500.]
第381行预测完成！
[557.]
第382行预测完成！
[352.]
第383行预测完成！
[226.]
第384行预测完成！
[199.]
第385行预测完成！
[207.]
第386行预测完成！
[211.]
第387行预测完成！
[201.]
第388行预测完成！
[192.]
第389行预测完成！
[217.]
第390行预测完成！
[473.]
第391行预测完成！
[473.]
第392行预测完成！
[430.]
第393行预测完成！
[378.]
第394行预测完成！
[608.]
第395行预测完成！
[798.]
第396行预测完成！
[560.]
第397行预测完成！
[429.]
第398行预测完成！
[411.]
第399行预测完成！
[388.]
第400行预测完成！
[726.]
第401行预测完成！
[695.]
第402行预测完成！
[482.]
第403行预测完成！
[417.]
第404行预测完成！
[493.]
第405行预测完成！
[545.]
第406行预测完成！
[366.]
第407行预测完成！
[223.]
第408行预测完成！
[199.]
第409行预测完成！
[198.]
第410行预测完成！
[198.]
第411行预测完成！
[195.]
第412行预测完成！
[185.]
第413行预测完成！
[211.]
第414行预测完成！
[485.]
第415行预测完成！
[488.]
第416行预测完成！
[427.]
第417行预测完成！
[368.]
第418行预测完成！
[603.]
第419行预测完成！
[773.]
第420行预测完成！
[540.]
第421行预测完成！
[426.]
第422行预测完成！
[401.]
第42

第286行预测完成！
[131.]
第287行预测完成！
[27.]
第288行预测完成！
[8.]
第289行预测完成！
[4.]
第290行预测完成！
[2.]
第291行预测完成！
[4.]
第292行预测完成！
[17.]
第293行预测完成！
[93.]
第294行预测完成！
[386.]
第295行预测完成！
[573.]
第296行预测完成！
[763.]
第297行预测完成！
[706.]
第298行预测完成！
[1047.]
第299行预测完成！
[987.]
第300行预测完成！
[747.]
第301行预测完成！
[618.]
第302行预测完成！
[675.]
第303行预测完成！
[707.]
第304行预测完成！
[1126.]
第305行预测完成！
[1018.]
第306行预测完成！
[631.]
第307行预测完成！
[509.]
第308行预测完成！
[467.]
第309行预测完成！
[378.]
第310行预测完成！
[124.]
第311行预测完成！
[27.]
第312行预测完成！
[8.]
第313行预测完成！
[4.]
第314行预测完成！
[3.]
第315行预测完成！
[3.]
第316行预测完成！
[11.]
第317行预测完成！
[63.]
第318行预测完成！
[229.]
第319行预测完成！
[383.]
第320行预测完成！
[526.]
第321行预测完成！
[592.]
第322行预测完成！
[803.]
第323行预测完成！
[812.]
第324行预测完成！
[597.]
第325行预测完成！
[512.]
第326行预测完成！
[553.]
第327行预测完成！
[637.]
第328行预测完成！
[950.]
第329行预测完成！
[836.]
第330行预测完成！
[598.]
第331行预测完成！
[491.]
第332行预测完成！
[446.]
第333行预测完成！
[371.]
第334行预测完成！
[149.]
第335行预测完成！
[37.]
第336行预测完成！
[17.]
第337行预测完成！
[13.]
第338行预测完成！
[14.]
第339行预测完成！
[15.]
第340行预测完成！
[28.]
第341行预测完成！
[99.]
第342行预测完成！
[448.]


第204行预测完成！
[911.]
第205行预测完成！
[764.]
第206行预测完成！
[708.]
第207行预测完成！
[658.]
第208行预测完成！
[1217.]
第209行预测完成！
[1274.]
第210行预测完成！
[838.]
第211行预测完成！
[801.]
第212行预测完成！
[847.]
第213行预测完成！
[832.]
第214行预测完成！
[372.]
第215行预测完成！
[162.]
第216行预测完成！
[132.]
第217行预测完成！
[118.]
第218行预测完成！
[116.]
第219行预测完成！
[111.]
第220行预测完成！
[114.]
第221行预测完成！
[217.]
第222行预测完成！
[719.]
第223行预测完成！
[965.]
第224行预测完成！
[783.]
第225行预测完成！
[592.]
第226行预测完成！
[1112.]
第227行预测完成！
[1413.]
第228行预测完成！
[922.]
第229行预测完成！
[777.]
第230行预测完成！
[726.]
第231行预测完成！
[663.]
第232行预测完成！
[1225.]
第233行预测完成！
[1265.]
第234行预测完成！
[847.]
第235行预测完成！
[789.]
第236行预测完成！
[856.]
第237行预测完成！
[843.]
第238行预测完成！
[364.]
第239行预测完成！
[140.]
第240行预测完成！
[118.]
第241行预测完成！
[114.]
第242行预测完成！
[112.]
第243行预测完成！
[114.]
第244行预测完成！
[117.]
第245行预测完成！
[221.]
第246行预测完成！
[718.]
第247行预测完成！
[945.]
第248行预测完成！
[780.]
第249行预测完成！
[608.]
第250行预测完成！
[1108.]
第251行预测完成！
[1449.]
第252行预测完成！
[934.]
第253行预测完成！
[829.]
第254行预测完成！
[754.]
第255行预测完成！
[674.]
第256行预测完成！
[1220.]
第257行预测完成！
[1213.]
第258行预测完成！
[818.]


第111行预测完成！
[361.]
第112行预测完成！
[626.]
第113行预测完成！
[621.]
第114行预测完成！
[423.]
第115行预测完成！
[377.]
第116行预测完成！
[386.]
第117行预测完成！
[301.]
第118行预测完成！
[149.]
第119行预测完成！
[37.]
第120行预测完成！
[12.]
第121行预测完成！
[6.]
第122行预测完成！
[4.]
第123行预测完成！
[6.]
第124行预测完成！
[15.]
第125行预测完成！
[64.]
第126行预测完成！
[254.]
第127行预测完成！
[460.]
第128行预测完成！
[390.]
第129行预测完成！
[320.]
第130行预测完成！
[447.]
第131行预测完成！
[605.]
第132行预测完成！
[428.]
第133行预测完成！
[336.]
第134行预测完成！
[332.]
第135行预测完成！
[350.]
第136行预测完成！
[571.]
第137行预测完成！
[527.]
第138行预测完成！
[377.]
第139行预测完成！
[345.]
第140行预测完成！
[343.]
第141行预测完成！
[277.]
第142行预测完成！
[130.]
第143行预测完成！
[33.]
第144行预测完成！
[13.]
第145行预测完成！
[6.]
第146行预测完成！
[4.]
第147行预测完成！
[3.]
第148行预测完成！
[8.]
第149行预测完成！
[55.]
第150行预测完成！
[202.]
第151行预测完成！
[370.]
第152行预测完成！
[360.]
第153行预测完成！
[325.]
第154行预测完成！
[509.]
第155行预测完成！
[589.]
第156行预测完成！
[409.]
第157行预测完成！
[329.]
第158行预测完成！
[324.]
第159行预测完成！
[349.]
第160行预测完成！
[571.]
第161行预测完成！
[529.]
第162行预测完成！
[390.]
第163行预测完成！
[314.]
第164行预测完成！
[306.]
第165行预测完成！
[275.]
第166行预测完成！
[121.]
第167行预测完成！
[3

第31行预测完成！
[775.]
第32行预测完成！
[895.]
第33行预测完成！
[786.]
第34行预测完成！
[915.]
第35行预测完成！
[799.]
第36行预测完成！
[847.]
第37行预测完成！
[740.]
第38行预测完成！
[792.]
第39行预测完成！
[743.]
第40行预测完成！
[975.]
第41行预测完成！
[780.]
第42行预测完成！
[571.]
第43行预测完成！
[565.]
第44行预测完成！
[457.]
第45行预测完成！
[328.]
第46行预测完成！
[114.]
第47行预测完成！
[32.]
第48行预测完成！
[16.]
第49行预测完成！
[14.]
第50行预测完成！
[16.]
第51行预测完成！
[18.]
第52行预测完成！
[35.]
第53行预测完成！
[86.]
第54行预测完成！
[461.]
第55行预测完成！
[748.]
第56行预测完成！
[901.]
第57行预测完成！
[778.]
第58行预测完成！
[935.]
第59行预测完成！
[803.]
第60行预测完成！
[804.]
第61行预测完成！
[758.]
第62行预测完成！
[884.]
第63行预测完成！
[810.]
第64行预测完成！
[1081.]
第65行预测完成！
[824.]
第66行预测完成！
[570.]
第67行预测完成！
[609.]
第68行预测完成！
[487.]
第69行预测完成！
[344.]
第70行预测完成！
[116.]
第71行预测完成！
[29.]
第72行预测完成！
[15.]
第73行预测完成！
[13.]
第74行预测完成！
[13.]
第75行预测完成！
[14.]
第76行预测完成！
[30.]
第77行预测完成！
[85.]
第78行预测完成！
[463.]
第79行预测完成！
[701.]
第80行预测完成！
[906.]
第81行预测完成！
[779.]
第82行预测完成！
[956.]
第83行预测完成！
[786.]
第84行预测完成！
[806.]
第85行预测完成！
[766.]
第86行预测完成！
[795.]
第87行预测完成！
[808.]
第88行预测完成！
[1008.]
第89行预测完成！
[783.]
第90行预测完成！

第509行预测完成！
[87.]
第510行预测完成！
[372.]
第511行预测完成！
[698.]
第512行预测完成！
[707.]
第513行预测完成！
[693.]
第514行预测完成！
[918.]
第515行预测完成！
[865.]
第516行预测完成！
[695.]
第517行预测完成！
[705.]
第518行预测完成！
[737.]
第519行预测完成！
[818.]
第520行预测完成！
[1093.]
第521行预测完成！
[880.]
第522行预测完成！
[592.]
第523行预测完成！
[555.]
第524行预测完成！
[500.]
第525行预测完成！
[384.]
第526行预测完成！
[135.]
第527行预测完成！
[30.]
第528行预测完成！
[14.]
第529行预测完成！
[10.]
第530行预测完成！
[10.]
第531行预测完成！
[11.]
第532行预测完成！
[24.]
第533行预测完成！
[80.]
第534行预测完成！
[485.]
第535行预测完成！
[829.]
第536行预测完成！
[777.]
第537行预测完成！
[757.]
第538行预测完成！
[1041.]
第539行预测完成！
[900.]
第540行预测完成！
[731.]
第541行预测完成！
[754.]
第542行预测完成！
[800.]
第543行预测完成！
[832.]
第544行预测完成！
[1179.]
第545行预测完成！
[903.]
第546行预测完成！
[574.]
第547行预测完成！
[578.]
第548行预测完成！
[529.]
第549行预测完成！
[390.]
第550行预测完成！
[137.]
第551行预测完成！
******删减前地点数据大小：1272******
******删减后地点数据大小：552******
[145.]
第0行预测完成！
[82.]
第1行预测完成！
[51.]
第2行预测完成！
[48.]
第3行预测完成！
[49.]
第4行预测完成！
[71.]
第5行预测完成！
[132.]
第6行预测完成！
[413.]
第7行预测完成！
[694.]
第8行预测完成！
[693.]
第9行预测完成！
[539.]
第10行预测完成！
[962.]
第11行预测

[774.]
第430行预测完成！
[361.]
第431行预测完成！
[99.]
第432行预测完成！
[54.]
第433行预测完成！
[43.]
第434行预测完成！
[39.]
第435行预测完成！
[36.]
第436行预测完成！
[54.]
第437行预测完成！
[141.]
第438行预测完成！
[613.]
第439行预测完成！
[899.]
第440行预测完成！
[710.]
第441行预测完成！
[538.]
第442行预测完成！
[953.]
第443行预测完成！
[1154.]
第444行预测完成！
[799.]
第445行预测完成！
[687.]
第446行预测完成！
[622.]
第447行预测完成！
[582.]
第448行预测完成！
[1004.]
第449行预测完成！
[942.]
第450行预测完成！
[619.]
第451行预测完成！
[608.]
第452行预测完成！
[666.]
第453行预测完成！
[691.]
第454行预测完成！
[352.]
第455行预测完成！
[117.]
第456行预测完成！
[57.]
第457行预测完成！
[42.]
第458行预测完成！
[35.]
第459行预测完成！
[33.]
第460行预测完成！
[46.]
第461行预测完成！
[121.]
第462行预测完成！
[348.]
第463行预测完成！
[646.]
第464行预测完成！
[661.]
第465行预测完成！
[533.]
第466行预测完成！
[855.]
第467行预测完成！
[969.]
第468行预测完成！
[695.]
第469行预测完成！
[599.]
第470行预测完成！
[573.]
第471行预测完成！
[567.]
第472行预测完成！
[819.]
第473行预测完成！
[805.]
第474行预测完成！
[605.]
第475行预测完成！
[535.]
第476行预测完成！
[532.]
第477行预测完成！
[593.]
第478行预测完成！
[346.]
第479行预测完成！
[125.]
第480行预测完成！
[62.]
第481行预测完成！
[44.]
第482行预测完成！
[35.]
第483行预测完成！
[31.]
第484行预测完成！
[39.]
第485行预测完成！
[109.]

第338行预测完成！
[12.]
第339行预测完成！
[15.]
第340行预测完成！
[45.]
第341行预测完成！
[228.]
第342行预测完成！
[1595.]
第343行预测完成！
[2530.]
第344行预测完成！
[1485.]
第345行预测完成！
[1093.]
第346行预测完成！
[3910.]
第347行预测完成！
[4539.]
第348行预测完成！
[1957.]
第349行预测完成！
[1262.]
第350行预测完成！
[1173.]
第351行预测完成！
[1294.]
第352行预测完成！
[3741.]
第353行预测完成！
[3364.]
第354行预测完成！
[2000.]
第355行预测完成！
[1431.]
第356行预测完成！
[1132.]
第357行预测完成！
[729.]
第358行预测完成！
[259.]
第359行预测完成！
[39.]
第360行预测完成！
[12.]
第361行预测完成！
[6.]
第362行预测完成！
[8.]
第363行预测完成！
[13.]
第364行预测完成！
[41.]
第365行预测完成！
[218.]
第366行预测完成！
[1738.]
第367行预测完成！
[2558.]
第368行预测完成！
[1593.]
第369行预测完成！
[1100.]
第370行预测完成！
[3898.]
第371行预测完成！
[4707.]
第372行预测完成！
[1969.]
第373行预测完成！
[1090.]
第374行预测完成！
[1047.]
第375行预测完成！
[1140.]
第376行预测完成！
[3489.]
第377行预测完成！
[3239.]
第378行预测完成！
[1911.]
第379行预测完成！
[1366.]
第380行预测完成！
[1109.]
第381行预测完成！
[711.]
第382行预测完成！
[247.]
第383行预测完成！
[42.]
第384行预测完成！
[13.]
第385行预测完成！
[6.]
第386行预测完成！
[7.]
第387行预测完成！
[11.]
第388行预测完成！
[37.]
第389行预测完成！
[202.]
第390行预测完成！
[1756.]
第391行预测完成！
[2666.]
第392行预测完成！
[157

第246行预测完成！
[761.]
第247行预测完成！
[999.]
第248行预测完成！
[868.]
第249行预测完成！
[711.]
第250行预测完成！
[1066.]
第251行预测完成！
[1132.]
第252行预测完成！
[896.]
第253行预测完成！
[737.]
第254行预测完成！
[637.]
第255行预测完成！
[664.]
第256行预测完成！
[990.]
第257行预测完成！
[981.]
第258行预测完成！
[695.]
第259行预测完成！
[635.]
第260行预测完成！
[688.]
第261行预测完成！
[743.]
第262行预测完成！
[538.]
第263行预测完成！
[451.]
第264行预测完成！
[397.]
第265行预测完成！
[363.]
第266行预测完成！
[357.]
第267行预测完成！
[352.]
第268行预测完成！
[352.]
第269行预测完成！
[437.]
第270行预测完成！
[732.]
第271行预测完成！
[974.]
第272行预测完成！
[829.]
第273行预测完成！
[685.]
第274行预测完成！
[1003.]
第275行预测完成！
[1077.]
第276行预测完成！
[859.]
第277行预测完成！
[728.]
第278行预测完成！
[642.]
第279行预测完成！
[646.]
第280行预测完成！
[917.]
第281行预测完成！
[881.]
第282行预测完成！
[621.]
第283行预测完成！
[575.]
第284行预测完成！
[600.]
第285行预测完成！
[705.]
第286行预测完成！
[496.]
第287行预测完成！
[403.]
第288行预测完成！
[387.]
第289行预测完成！
[387.]
第290行预测完成！
[395.]
第291行预测完成！
[374.]
第292行预测完成！
[375.]
第293行预测完成！
[455.]
第294行预测完成！
[654.]
第295行预测完成！
[867.]
第296行预测完成！
[780.]
第297行预测完成！
[661.]
第298行预测完成！
[891.]
第299行预测完成！
[958.]
第300行预测完成！
[802.]
第301行预

第149行预测完成！
[162.]
第150行预测完成！
[497.]
第151行预测完成！
[778.]
第152行预测完成！
[871.]
第153行预测完成！
[847.]
第154行预测完成！
[1394.]
第155行预测完成！
[1653.]
第156行预测完成！
[1186.]
第157行预测完成！
[900.]
第158行预测完成！
[811.]
第159行预测完成！
[866.]
第160行预测完成！
[1313.]
第161行预测完成！
[1488.]
第162行预测完成！
[1107.]
第163行预测完成！
[903.]
第164行预测完成！
[859.]
第165行预测完成！
[840.]
第166行预测完成！
[356.]
第167行预测完成！
[77.]
第168行预测完成！
[41.]
第169行预测完成！
[41.]
第170行预测完成！
[39.]
第171行预测完成！
[39.]
第172行预测完成！
[54.]
第173行预测完成！
[170.]
第174行预测完成！
[803.]
第175行预测完成！
[1188.]
第176行预测完成！
[1118.]
第177行预测完成！
[990.]
第178行预测完成！
[1805.]
第179行预测完成！
[2277.]
第180行预测完成！
[1558.]
第181行预测完成！
[1162.]
第182行预测完成！
[1144.]
第183行预测完成！
[1180.]
第184行预测完成！
[1853.]
第185行预测完成！
[1871.]
第186行预测完成！
[1326.]
第187行预测完成！
[1146.]
第188行预测完成！
[1160.]
第189行预测完成！
[1159.]
第190行预测完成！
[454.]
第191行预测完成！
[83.]
第192行预测完成！
[45.]
第193行预测完成！
[33.]
第194行预测完成！
[27.]
第195行预测完成！
[26.]
第196行预测完成！
[56.]
第197行预测完成！
[200.]
第198行预测完成！
[1142.]
第199行预测完成！
[1425.]
第200行预测完成！
[1199.]
第201行预测完成！
[968.]
第202行预测完成！
[1766.]
第203行预测完成！
[2379

第56行预测完成！
[490.]
第57行预测完成！
[414.]
第58行预测完成！
[684.]
第59行预测完成！
[949.]
第60行预测完成！
[654.]
第61行预测完成！
[538.]
第62行预测完成！
[478.]
第63行预测完成！
[443.]
第64行预测完成！
[757.]
第65行预测完成！
[784.]
第66行预测完成！
[523.]
第67行预测完成！
[420.]
第68行预测完成！
[442.]
第69行预测完成！
[456.]
第70行预测完成！
[235.]
第71行预测完成！
[120.]
第72行预测完成！
[96.]
第73行预测完成！
[86.]
第74行预测完成！
[84.]
第75行预测完成！
[86.]
第76行预测完成！
[95.]
第77行预测完成！
[145.]
第78行预测完成！
[463.]
第79行预测完成！
[610.]
第80行预测完成！
[499.]
第81行预测完成！
[393.]
第82行预测完成！
[674.]
第83行预测完成！
[964.]
第84行预测完成！
[638.]
第85行预测完成！
[508.]
第86行预测完成！
[491.]
第87行预测完成！
[439.]
第88行预测完成！
[777.]
第89行预测完成！
[808.]
第90行预测完成！
[520.]
第91行预测完成！
[439.]
第92行预测完成！
[460.]
第93行预测完成！
[474.]
第94行预测完成！
[249.]
第95行预测完成！
[133.]
第96行预测完成！
[111.]
第97行预测完成！
[110.]
第98行预测完成！
[104.]
第99行预测完成！
[102.]
第100行预测完成！
[108.]
第101行预测完成！
[157.]
第102行预测完成！
[458.]
第103行预测完成！
[571.]
第104行预测完成！
[488.]
第105行预测完成！
[378.]
第106行预测完成！
[622.]
第107行预测完成！
[881.]
第108行预测完成！
[612.]
第109行预测完成！
[483.]
第110行预测完成！
[466.]
第111行预测完成！
[421.]
第112行预测完成！
[734.]
第113行预测完成！
[741.]
第114行

[881.]
第522行预测完成！
[587.]
第523行预测完成！
[493.]
第524行预测完成！
[515.]
第525行预测完成！
[535.]
第526行预测完成！
[289.]
第527行预测完成！
[128.]
第528行预测完成！
[105.]
第529行预测完成！
[98.]
第530行预测完成！
[96.]
第531行预测完成！
[90.]
第532行预测完成！
[95.]
第533行预测完成！
[140.]
第534行预测完成！
[485.]
第535行预测完成！
[645.]
第536行预测完成！
[518.]
第537行预测完成！
[383.]
第538行预测完成！
[751.]
第539行预测完成！
[1010.]
第540行预测完成！
[696.]
第541行预测完成！
[552.]
第542行预测完成！
[471.]
第543行预测完成！
[453.]
第544行预测完成！
[887.]
第545行预测完成！
[871.]
第546行预测完成！
[594.]
第547行预测完成！
[499.]
第548行预测完成！
[515.]
第549行预测完成！
[521.]
第550行预测完成！
[284.]
第551行预测完成！
******删减前地点数据大小：1272******
******删减后地点数据大小：552******
[18.]
第0行预测完成！
[19.]
第1行预测完成！
[20.]
第2行预测完成！
[22.]
第3行预测完成！
[25.]
第4行预测完成！
[32.]
第5行预测完成！
[65.]
第6行预测完成！
[231.]
第7行预测完成！
[429.]
第8行预测完成！
[552.]
第9行预测完成！
[504.]
第10行预测完成！
[579.]
第11行预测完成！
[569.]
第12行预测完成！
[545.]
第13行预测完成！
[480.]
第14行预测完成！
[559.]
第15行预测完成！
[587.]
第16行预测完成！
[825.]
第17行预测完成！
[666.]
第18行预测完成！
[466.]
第19行预测完成！
[476.]
第20行预测完成！
[412.]
第21行预测完成！
[313.]
第22行预测完成！
[111.]
第23行预测完成！
[32.]
第24行预测完成！
[17

第439行预测完成！
[774.]
第440行预测完成！
[785.]
第441行预测完成！
[641.]
第442行预测完成！
[919.]
第443行预测完成！
[797.]
第444行预测完成！
[650.]
第445行预测完成！
[611.]
第446行预测完成！
[675.]
第447行预测完成！
[721.]
第448行预测完成！
[944.]
第449行预测完成！
[743.]
第450行预测完成！
[500.]
第451行预测完成！
[460.]
第452行预测完成！
[423.]
第453行预测完成！
[314.]
第454行预测完成！
[99.]
第455行预测完成！
[24.]
第456行预测完成！
[9.]
第457行预测完成！
[5.]
第458行预测完成！
[3.]
第459行预测完成！
[3.]
第460行预测完成！
[11.]
第461行预测完成！
[50.]
第462行预测完成！
[239.]
第463行预测完成！
[446.]
第464行预测完成！
[631.]
第465行预测完成！
[574.]
第466行预测完成！
[744.]
第467行预测完成！
[734.]
第468行预测完成！
[541.]
第469行预测完成！
[503.]
第470行预测完成！
[532.]
第471行预测完成！
[573.]
第472行预测完成！
[758.]
第473行预测完成！
[657.]
第474行预测完成！
[460.]
第475行预测完成！
[398.]
第476行预测完成！
[357.]
第477行预测完成！
[286.]
第478行预测完成！
[78.]
第479行预测完成！
[16.]
第480行预测完成！
[7.]
第481行预测完成！
[3.]
第482行预测完成！
[2.]
第483行预测完成！
[1.]
第484行预测完成！
[6.]
第485行预测完成！
[36.]
第486行预测完成！
[142.]
第487行预测完成！
[290.]
第488行预测完成！
[399.]
第489行预测完成！
[452.]
第490行预测完成！
[581.]
第491行预测完成！
[628.]
第492行预测完成！
[428.]
第493行预测完成！
[373.]
第494行预测完成！
[373.]
第495行预测完成！
[424.]

In [49]:
pred = pred_df.num_of_people
real = test.num_of_people
rmse = np.sqrt(mean_squared_error(pred,real))
print('Test RMSE:',rmse)

Test RMSE: 323.7340279454762


## 实验记录
1. 0填充 训练数据从2月份开始 rmse：358
2. 前一个非null值填充 去掉二月数据 异常值处理 rmse：377
3. 前一个非null值填充 去掉二月数据 rmse：412
4. 0填充 训练数据从3月份开始 rmse：378
5. 按地点前一时刻非空值填充 训练数据从4月份开始 rmse：576
6. 0填充 训练数据从4月份开始 去掉十一 rmse：582
7. 0填充 训练数据从2月份开始 去掉十一 增加小时特征 rmse：323

# 公共函数

In [12]:
# 按地点分组，加快处理速度
def locID_split(data):
    res = []
    df = data.copy()
    for i in df.loc_id.unique():
        tmp = df[df.loc_id == i]
        tmp.reset_index(inplace=True) # ！！！注意重置索引 实时预测使用的loc方法，行索引重置后才能正确填充
        tmp.drop('index',inplace=True,axis=1)
        res.append(tmp)
    return res

In [13]:
# 实时预测填充
# 思路：第一行特征完整，进行预测，预测出来的价格添加到后边time_stamp对应的历史特征上去，接着预测第二行，进行迭代
def real_time_predict(model,data):
    df = data.copy()
    test_list = locID_split(df)
    res = pd.DataFrame()
    for tmp in test_list:
        for i in range(tmp.shape[0]):
            X = xgb.DMatrix(tmp.iloc[i:i+1,3:])
            pred = np.round(model.predict(X,ntree_limit = model.best_iteration))
            print(pred)
            tmp.loc[i,'num_of_people'] = pred
            for j in range(len(columns_24h)):
                tmp.loc[i+j+1,columns_24h[j]] = pred
            for k in range(len(columns_30d)):
                tmp.loc[i+(k+1)*24,columns_30d[k]] = pred
            print('第%s行预测完成！'%i)
        print('******删减前地点数据大小：%s******'%tmp.shape[0])
        tmp.dropna(axis=0,inplace=True) # 数据在填充历史特征时，会多填充不需要的数据，去掉
        print('******删减后地点数据大小：%s******'%tmp.shape[0])
        res = res.append(tmp) # df.append返回DataFrame，注意赋值
    return res

In [14]:
reg = re.compile('\s+|\t|\n') # 正则匹配空格、Tab键、换行
#列名，历史24小时人数
col_24h='''history_1h_population,history_2h_population,history_3h_population,history_4h_population
    ,history_5h_population,history_6h_population,history_7h_population,history_8h_population
    ,history_9h_population,history_10h_population,history_11h_population,history_12h_population
    ,history_13h_population,history_14h_population,history_15h_population,history_16h_population,history_17h_population
    ,history_18h_population,history_19h_population,history_20h_population,history_21h_population,history_22h_population
    ,history_23h_population'''
columns_24h=re.sub(reg,'',col_24h).split(',')  
#列名，同一时刻历史15天人数
col_30d='''history_1d_population,history_2d_population,history_3d_population,
    history_4d_population,history_5d_population,history_6d_population,
    history_7d_population,history_8d_population,history_9d_population,history_10d_population,history_11d_population,
    history_12d_population,history_13d_population,history_14d_population,history_15d_population,history_16d_population
    ,history_17d_population,history_18d_population,history_19d_population,history_20d_population,history_21d_population
    ,history_22d_population,history_23d_population,history_24d_population,history_25d_population,history_26d_population
    ,history_27d_population,history_28d_population,history_29d_population,history_30d_population'''
columns_30d=re.sub(reg,'',col_30d).split(',')  

In [15]:
def xgboost_model_train(train_X,train_y,test_X,test_y):
    # 模型训练
    xgb_train  = xgb.DMatrix(train_X, train_y)
    xgb_test   = xgb.DMatrix(test_X, test_y)
    params_xgb = {
#         'booster': 'gbtree',                # 取决于使用哪种booster,有两中模型可以选择gbtree和gblinear。gbtree使用基于树的模型进行提升计算，gblinear使用线性模型进行提升计算
        'objective': 'reg:linear',     # 定义学习任务及相应的学习目标，可选的目标函数如下：“reg:linear” —— 线性回归,“reg:logistic”—— 逻辑回归。
        #'multi:softmax'
        'gamma': 0.03,                      # 最小损失减少，模型在默认情况下，对于一个节点的划分只有在其loss function 得到结果大于0的情况下才进行，而gamma 给定了所需的最低loss function的值
        'max_depth': 8,                     # 数的最大深度。缺省值为6，树的深度越大，则对数据的拟合程度越高（过拟合程度也越高）。即该参数也是控制过拟合
        'lambda': 4,                        # L2 正则的惩罚系数
        'subsample': 0.8,                   # 用于训练模型的子样本占整个样本集合的比例。如果设置为0.5则意味着XGBoost将随机的从整个样本集合中随机的抽取出50%的子样本建立树模型，这能够防止过拟合。
        'colsample_bytree': 0.8,            # 在建立树时对特征采样的比例。缺省值为1。
#         'eval_metric':'mape_ln',
        'min_child_weight': 8.5,            # 孩子节点中最小的样本权重和，在现行回归模型中，这个参数是指建立每个模型所需要的最小样本数。该成熟越大算法越conservative。即调大这个参数能够控制过拟合。
        'silent': 1,                        # 取0时表示打印出运行时信息，取1时表示以缄默方式运行，不打印运行时信息。缺省值为0。
        'eta': 0.03,                        # 为了防止过拟合，更新过程中用到的收缩步长。通常最后设置eta为0.01~0.2
        'seed': 1000}                       # 随机数的种子。缺省值为0。
    num_boost_round = 1000
    plst  = params_xgb.items()
    model_xgb = xgb.train(plst, xgb_train, num_boost_round,evals=[(xgb_train,'train'),(xgb_test,'test')],early_stopping_rounds=20,verbose_eval=True)
    return model_xgb